### ISSUE TITLE
projectNameQuestion should always prompt during amplify init

### ISSUE THREAD

**Issue by hakanson** [aws-amplify/amplify-cli#662]
**Is your feature request related to a problem? Please describe.**
I want to control the naming of AWS resources, and `projectName` from `project-config.json` is used as the default prefix on many AWS resources created during `amplify init`.  However, I am frustrated with the default value used (as my specific project/account enforces some naming standards).

**Describe the solution you'd like**
I noticed in [s0-analyzeProject.js L9](https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-cli/src/lib/init-steps/s0-analyzeProject.js#L9) that the project name is determined from project directory and there is not an option to change unless it is either too long or too short.
```
    let projectName = path.basename(projectPath);
    projectName = projectName.replace(/[^a-zA-Z0-9]/g, '');

    if (projectName.length > 20 || projectName.length < 3) {
      const projectNameQuestion = {
        type: 'input',
        name: 'projectName',
        message: 'Enter a name for the project',
        validate: input => new Promise((resolvePromise, reject) => ((input.length > 20 || input.length < 3 || /[^a-zA-Z0-9]/g.test(input)) ? reject(new Error('Project name should be between 3 and 20 characters and alphanumeric')) : resolvePromise(true))),
      };
```
Oddly, [c0-analyzeProject.js L47](https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-cli/src/lib/config-steps/c0-analyzeProject.js#L47) prompts with a default and I think that behavior should be copied.

```
default: projectName,
```

**Describe alternatives you've considered**
I've renamed my parent directory to be the name I needed, but that wasn't an obvious solution until I looked into the code.  I am also waiting on #574 for full CF customization.

**Additional context**
None.

**Comment by kaustavghosh06** [comment]
@hakanson We realized that and we prompt for the projectName as a part of the beta @multienv version of the CLI. It'll get merged into master soon.
